mlserver/grpc/interceptors.py

```diff
@@ -99,60 +66,111 @@ def _metrics_wrapper(
                         self._interceptor._metrics[
-                            "grpc_server_started_counter"
+                            "grpc_server_handled_histogram"
                         ].labels(
                             grpc_type=grpc_type,
                             grpc_service=grpc_service_name,
                             grpc_method=grpc_method_name,
-                        ).inc()
+                        ).observe(
+                            max(default_timer() - start, 0)
+                        )
+            except Exception as e:  # pylint: disable=broad-except
+                # Allow user to skip the exceptions in order to maintain
+                # the basic functionality in the server
+                # The logging function in exception can be toggled with log_exceptions
+                # in order to suppress the noise in logging
+                if self._interceptor._skip_exceptions:
+                    if self._interceptor._log_exceptions:
+                        logger.error(e)
 
-                    # Invoke the original rpc behavior.
-                    # NOTE: This is the main change required with respect to
-                    # the original implementation in `py-grpc-prometheus`.
-                    response_or_iterator = await old_handler(
-                        request_or_iterator, servicer_context
+                    if response is None:
+                        return response
+
+                    return await behavior(request, servicer_context)
+                raise e
+
+        async def new_behavior_stream(
+            request_async_iterator, servicer_context: ServicerContext
+        ):
+            response_async_iterator = None
+            try:
+                grpc_type = grpc_utils.get_method_type(
+                    request_streaming, response_streaming
+                )
+                try:
+                    request_async_iterator = wrap_async_iterator_inc_counter(
+                        request_async_iterator,
+                        self._interceptor._metrics["grpc_server_stream_msg_received"],
+                        grpc_type,
+                        grpc_service_name,
+                        grpc_method_name,
                     )
 
-                    if response_streaming:
-                        sent_metric = self._interceptor._metrics[
-                            "grpc_server_stream_msg_sent"
-                        ]
-                        response_or_iterator = grpc_utils.wrap_iterator_inc_counter(
-                            response_or_iterator,
-                            sent_metric,
-                            grpc_type,
-                            grpc_service_name,
-                            grpc_method_name,
-                        )
+                    # wrap the original behavior with the metrics
+                    sent_metric = self._interceptor._metrics[
```

Comment:
nit: the above code uses `self._interceptor._metrics["grpc_server_stream_msg_received"]` directly, while the below code doesnt. Is this to get around line width? perhaps unify style?